src/vorta/borg/borg_thread.py

```diff
@@ -13,30 +13,39 @@
 from PyQt5.QtWidgets import QApplication
 from subprocess import Popen, PIPE, TimeoutExpired
 
+from vorta.borg.job_scheduler import Job, DEBUG, JobStatus
 from vorta.i18n import trans_late, translate
 from vorta.models import EventLogModel, BackupProfileMixin
 from vorta.utils import borg_compat, pretty_bytes
 from vorta.keyring.abc import VortaKeyring
 from vorta.keyring.db import VortaDBKeyring
 
-mutex = Lock()
+temp_mutex = Lock()
 logger = logging.getLogger(__name__)
 
 FakeRepo = namedtuple('Repo', ['url', 'id', 'extra_borg_arguments', 'encryption'])
 FakeProfile = namedtuple('FakeProfile', ['repo', 'name', 'ssh_key'])
 
+"""
+All methods in this class must be thread safe. Particularly,
+I strongly unadvised global variable and class variables.
+Sqlite access are thread-safe because peewee is thread-safe.
+The method prepare is not thread-safe because of keyring and I don't know why. That's why I added a
+temporary mutex.
+"""
+
 
-class BorgThread(QtCore.QThread, BackupProfileMixin):
+class BorgJob(Job, BackupProfileMixin):
```

Comment:
The file should be called `borg_job.py` now?

Reply:
It makes sense. Its true that BorgJob class inherits from Job but Job is just an object, not a thread. You have to add it to the JobsManager to be run.

Reply:
Done locally.